src/ucp/core/ucp_worker.h

```diff
@@ -246,9 +252,14 @@ typedef struct ucp_worker {
     UCS_STATS_NODE_DECLARE(tm_offload_stats)
 
     ucs_cpu_set_t                 cpu_mask;        /* Save CPU mask for subsequent calls to ucp_worker_listen */
-    unsigned                      ep_config_max;   /* Maximal number of configurations */
-    unsigned                      ep_config_count; /* Current number of configurations */
-    ucp_ep_config_t               ep_config[0];    /* Array of transport limits and thresholds */
+
+    ucp_worker_rkey_config_hash_t rkey_config_hash; /* rkey config key -> index */
+
+    unsigned                      ep_config_count; /* Current number of ep configurations */
+    ucp_ep_config_t               ep_config[UCP_WORKER_MAX_CONFIG];
+
+    unsigned                      rkey_config_count; /* Current number of rkey configurations */
+    ucp_rkey_config_t             rkey_config[UCP_WORKER_MAX_CONFIG];
```

Comment:
maybe better define config arrays as pointers and allocate some reasonable space (not large) by default? Then configs maybe extended in run-time if needed.

Reply:
@brminich i wanted to avoid extra pointer dereference for reading ep configuration, on other hand since those arrays are quite large it'll be a different cache line anyway, WDYT?

Reply:
My concern is: Some users (shmem) may want to create many workers, but would not need many keys (no GPUs, etc). Would be nice not to consume many unneeded memory for configs per every worker.
Or, maybe we can define different (smaller) constant for number of ep configs (which is larger by size than rkey configs, so seems to be more important)

Reply:
@brminich fixed